pkg/controller/cluster/main-controller.go

```diff
@@ -352,31 +353,32 @@ func (c *Controller) syncHandler(key string) error {
 			runtime.HandleError(fmt.Errorf("Tenant '%s' in work queue no longer exists", key))
 			return nil
 		}
-		return err
+		return nil
 	}
-
 	// Set any required default values and init Global variables
 	nsName := types.NamespacedName{Namespace: namespace, Name: name}
-	mi.EnsureDefaults()
+	mi = mi.EnsureDefaults()
```

Comment:
Why the assignment? The function modifies the receiver

Reply:
we return the instance itself from that function. 

There is a bug in the present code where the defaults would get lost out of nowhere, it was caused by the status update, which returns an updated tenant object.